contracts/citycoin.clar

```diff
@@ -954,25 +960,34 @@ u113 u114 u115 u116 u117 u118 u119 u120 u121 u122 u123 u124 u125 u126 u127 u128
 ;; they locked up).
 (define-public (claim-stacking-reward (target-reward-cycle uint))
     (let (
-        (stacker tx-sender)    
-        (entitled-ustx (get-entitled-stacking-reward tx-sender target-reward-cycle block-height))
+        (stacker tx-sender)
+        (cur-reward-cycle (unwrap! (get-reward-cycle block-height) (err ERR-STACKING-NOT-AVAILABLE)))
+        (entitled-ustx (get-stacking-reward tx-sender target-reward-cycle))
         (to-return (get to-return (default-to { amount-token: u0, to-return: u0 } (get-stacked-per-cycle stacker target-reward-cycle))))
     )
     (begin
-        (asserts! (> entitled-ustx u0)
+        ;; check that target reward cycle is complete
+        (asserts! (> cur-reward-cycle target-reward-cycle)
+            (err ERR-CYCLE-NOT-COMPLETED))
+
+        ;; check that there are stacked tokens to redeem
+        (asserts! (> to-return u0)
             (err ERR-NOTHING-TO-REDEEM))
```

Comment:
With error thrown when `to-return` is not greater than `0` users won't be able to claim all of their stacking rewards.

When user stack over multiple cycles, then `to-return` is set to `0` for all cycles in locking period but the last one.
It means, with this logic if user stack over 32 cycles, he will be able to claim stacking reward only for the last cycle. If he try to claim reward for cycles 1-31 contract will throw an error `ERR-NOTHING-TO-REDEEM` and in many cases it will be a false statement.

Probably we should check here if user stacked in this cycle, and if he didn't throw appropriate error.

Reply:
Waiting on this to be resolved for my review. Ping when ready.

Reply:
Should be fixed in the latest commit, instead of failing on one value being 0, it should fail if both are 0 because there is nothing to redeem. The `asserts!` is updated to catch that now.

I also added an if statement to both transfers, which may be overkill but I like the idea of checking if there is something to transfer before making a transfer. Working on some tests to walk through similar scenarios and test the outcome, however this did work for me via the console as expected.

```none
(contract-call? .citycoin set-mining-activation-threshold u1)
(contract-call? .citycoin register-miner none)
::advance_chain_tip 200
(contract-call? .citycoin mine-tokens u100 none)
::advance_chain_tip 200
(contract-call? .citycoin claim-mining-reward u200)
(contract-call? .citycoin stack-tokens u100000 u401 u30)
::advance_chain_tip 2000
(contract-call? .citycoin mine-tokens u100 none)
::advance_chain_tip 2100
(contract-call? .citycoin get-stacking-reward 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE u1)
(contract-call? .citycoin claim-stacking-reward u1)
(contract-call? .citycoin mine-tokens u100 none)
::advance_chain_tip 61500
(contract-call? .citycoin get-stacking-reward 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE u2)
(contract-call? .citycoin get-stacking-reward 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE u30)
(contract-call? .citycoin get-stacked-per-cycle 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE u30)
(contract-call? .citycoin claim-stacking-reward u2)
(contract-call? .citycoin claim-stacking-reward u30)
```